perses/annihilation/new_relative.py

```
@@ -0,0 +1,1213 @@
+
+        #prepare dicts of forces, which will be useful later
+        self._old_system_forces = {type(force).__name__ : force for force in self._old_system.getForces()}
+        self._new_system_forces = {type(force).__name__ : force for force in self._new_system.getForces()}
+
+        #get and store the nonbonded method from the system:
+        self._nonbonded_method = self._old_system_forces['NonbondedForce'].getNonbondedMethod()
+
+        #start by creating an empty system. This will become the hybrid system.
+        self._hybrid_system = openmm.System()
+
+        #begin by copying all particles in the old system to the hybrid system. Note that this does not copy the
+        #interactions. It does, however, copy the particle masses. In general, hybrid index and old index should be
+        #the same.
+        for particle_idx in range(self._topology_proposal.n_atoms_old):
+            particle_mass = self._old_system.getParticleMass(particle_idx)
+            hybrid_idx = self._hybrid_system.addParticle(particle_mass)
+            self._old_to_hybrid_map[particle_idx] = hybrid_idx
+
+            #If the particle index in question is mapped, make sure to add it to the new to hybrid map as well.
+            if particle_idx in self._topology_proposal.old_to_new_atom_map.keys():
+                particle_index_in_new_system = self._topology_proposal.old_to_new_atom_map[particle_idx]
+                self._new_to_hybrid_map[particle_index_in_new_system] = hybrid_idx
+
+        #Next, add the remaining unique atoms from the new system to the hybrid system and map accordingly.
+        #As before, this does not copy interactions, only particle indices and masses.
+        for particle_idx in self._topology_proposal.unique_new_atoms:
+            particle_mass = self._new_system.getParticleMass(particle_idx)
+            hybrid_idx = self._hybrid_system.addParticle(particle_mass)
+            self._new_to_hybrid_map[particle_idx] = hybrid_idx
+
```

Yes! We don't have to support virtual sites in the alchemical region (probably should throw an exception) but we should definitely support vsites elsewhere.